doc/vxlan/data_plane_VXLAN_HLD.md

```diff
@@ -0,0 +1,507 @@
+
+Schema:
+
+; New table
+; specifies mapping of remote VTEP to VNI
+
+key = VXLAN_DATAPLANE_VTEP_VNI:remote_vtep_name:map_name
+                  ; Remote VTEP name + map name
+; field = value
+vni = VNI id
+
+
+### 4. Modules Design and Flows
+
+Following configurations are required for enabling EVPN L2 service on SONiC device:
+
+Create VXLAN data plane tunnel.
+Configure L2VNI-VLAN mapping for the given NVO.
+Configure control plane with remote VTEPS and bind them to the same vni as source VTEP.
+
+#### 4.1 The corresponding CONFIG_DB entries are as follows
+
+VLAN|Vlan{{vlan_id}}
+    "vlanid" : {{vlan_id}}
+
+VXLAN_TUNNEL_TABLE|{{source_vtep_name}}
+    "src_ip" : {{ipv4_address}}
+
+VXLAN_TUNNEL_MAP|{{source_vtep_name}}|{{tunnel_map_name}}
+    "vni" : {{vni_id}}
+    "vlan" : {{vlan_id}}
+
+VXLAN_REMOTE_VTEP_TABLE|{{remote_vtep_name}}
+    "src_ip" : {{ipv4_address}}
+
+VXLAN_REMOTE_VTEP_VNI_TABLE|{{remote_vtep_name}}|{{tunnel_map_name}}
+    "vni" : {{vni_id}}
+
+#### 4.2 Example for CONFIG_DB configuration
+
+VLAN|Vlan10
+    "vlanid" : 10
+
+VXLAN_TUNNEL_TABLE|vtep1
+    "src_ip" : "1.1.1.1"
+
+VXLAN_TUNNEL_MAP|{{source_vtep_name}}|{{tunnel_map_name}}
+    "vni" : 1000
+    "vlan" : 10
+
+VXLAN_REMOTE_VTEP_TABLE|vtep2
+    "remote_ip" : 1.1.1.2
+
+VXLAN_REMOTE_VTEP_VNI_TABLE|vtep2|map_1000
+    "vni" : 1000
+
+
+
+### 4.3 Tunnel creation
+
+Current implementation, supports both P2P and P2MP VXLAN tunnels for L2 scenarios.
+    The difference between the two approaches is the way in which the remote end point flooding is done. In P2P tunnel based approach, for every end point, a P2P tunnel object is created in the hardware and the bridge port created with this tunnel object is added as a VLAN member to the VLAN. In P2MP tunnel based approach, for each remote end point, along with local P2MP tunnel bridge port is added as L2MC group member along for the L2MC group associated with the VLAN.
+
+To correctly detect received packet's entry VXLAN tunnel source interface, P2P tunnel must be used, so Vendor's SAI implementation must support P2P.
+
+Adding first entry to VXLAN_REMOTE_VTEP_TABLE will create in entry VXLAN_DATAPLANE_VTEP_TABLE using existing VXLAN tunnel source address.
+
+#### 4.3.1 P2P Tunnel creation
+   
+   Currently created on :
+   * IMET route rx
+   * IP Prefix route handled by VRF construct.
+
+   Will be also created on remote tunnel VTEP add.
+
+  - Adding remote tunnel map entry will create entry in VXLAN_DATAPLANE_VTEP_VNI_TABLE (similar to IMET route Rx in EVPN).
+
+- SAI objects related to tunnel are created.
+    - Tunnel SAI object is created with the mapper IDs created for the VTEP.
+- BridgePort SAI object with the tunnel oid is created. Learning is disabled for EVPN tunnels and enabled for data plane tunnels.
+- Port object of type Tunnel created in the portsorch.
+
+The data plane tunnels are created when the remote VTEP is added and will be deleted when the last remote VTEP is removed.
```

Comment:
I am unclear on the term "last" remote VTEP.
Wouldnt only one remote VTEP configuration be made ?

Reply:
There can be several remote VTEPs, each with it name and IP.